code/modules/library/library_equipment.dm

```diff
@@ -0,0 +1,412 @@
+#define BARCODE_MODE_SCAN_SELECT     1
+#define BARCODE_MODE_SCAN_INVENTORY  2
+#define BARCODE_MODE_CHECKOUT        3
+#define BARCODE_MODE_CHECKIN         4
+
+/*
+ * Bookcase
+ */
+
+/obj/structure/bookcase
+	name = "bookcase"
+	icon = 'icons/obj/library.dmi'
+	icon_state = "bookshelf-0"
+	anchored = TRUE
+	density = TRUE
+	opacity = TRUE
+	resistance_flags = FLAMMABLE
+	max_integrity = 200
+	armor = list(MELEE = 0, BULLET = 0, LASER = 0, ENERGY = 0, BOMB = 0, BIO = 0, RAD = 0, FIRE = 50, ACID = 0)
+	var/list/allowed_books = list(/obj/item/book, /obj/item/spellbook, /obj/item/storage/bible, /obj/item/tome) //Things allowed in the bookcase
+
+/obj/structure/bookcase/attackby(obj/item/O, mob/user)
```

Comment:
Don't need the `else`s down here due to you returning in the if statements scope

Reply:
changed out the elses